core/snmp/impl-snmp4j/src/main/java/org/opennms/netmgt/snmp/snmp4j/OpaqueExt.java

```diff
@@ -0,0 +1,192 @@
+        return getAdapter().getDouble();
+    }
+
+    @Override
+    public void setValue(OctetString value) {
+        super.setValue(value);
+        adapter = null;
+    }
+
+    @Override
+    public void decodeBER(BERInputStream inputStream) throws IOException {
+        super.decodeBER(inputStream);
+        adapter = null;
+    }
+
+    @Override
+    public int getSyntax() {
+        return SMIConstants.SYNTAX_OPAQUE;
+    }
+
+    public OpaqueValueType getValueType() {
+        return getAdapter().getValueType();
+    }
+
+    private static UnsupportedAdapter createUnsupportedAdapter(final byte[] bytes) {
+        OctetString octetString = new OctetString(bytes);
+        LOG.debug("Unsupported bytes provided to OpaqueExt {}", octetString.toString());
+        return new UnsupportedAdapter(octetString);
+    }
+
+    private static ErrorAdapter createErrorAdapter(String message, final byte[] bytes) {
+        if (LOG.isDebugEnabled()) {
+            OctetString octetString = new OctetString(bytes);
+            LOG.debug("Cannot create TypeAdapter. {} ;, data: {}", message, octetString.toString());
+        }
+        return new ErrorAdapter();
+    }
+
+    private static OpaqueTypeAdapter createAdapter(final byte[] bytes) {
+
+        if (bytes.length < 3) {
+            return createErrorAdapter("To short", bytes);
+        }
+        
+        //existing implementations here do not support extended type so the custom type retrieving is implemented
+        int offset;
+        byte type = (byte) (bytes[0] & ID_BITTS);
+        //when bit 6 == 0 then Ok otherwise compound value -> it will be not decoded
+        if ((type & BITTS_6) != 0) {
+            return createUnsupportedAdapter(bytes);
+        }
+
+        if (type != ID_BITTS) { //type is stored in bits 1-5 but only when <= 30
+            offset = 1;
+        } else { // otherwise it is stored in first 7 bits of second byte but only when <= 127
+            type = bytes[1];
+            if (type < BITTS_EX_TYPE) {
+                offset = 2;
+            } else { // otherwise too big type - not supported jet
+                return createUnsupportedAdapter(bytes);
+            }
+        }
+
+        ByteBuffer buffer = ByteBuffer.wrap(bytes, offset, bytes.length - offset);
+        try (BERInputStream is = new BERInputStream(buffer)) {
+            int length = BER.decodeLength(is);
+            
+            if (bytes.length - buffer.position() != length) { //remaining length must de eqoal to data length
+                return createErrorAdapter("Wrong length of wraped data", bytes);
+            }
+
+            switch (type) {
+                case 120:
+                    if (length == 4 ) { //check length
+                        float floatValue = buffer.getFloat();
+                        return new DoubleAdapter((double)floatValue);
+                    }
+                    break;
+                default:
+                    break;
```

Comment:
If the default is empty why to add this?

Reply:
To explicitly show  there is nothing more to do here (till new format need to be parsed)

Reply:
should I put additionally/instead of comments?

Reply:
I suggest using a comment rather than an extra block.